test/httpClient.test.js

```diff
@@ -63,6 +63,25 @@ test('client calls a https server twice', (t) => {
   })
 })
 
+test('client calculates correct duration when using pipelining', (t) => {
+  t.plan(4)
+
+  const lazyServer = helper.startServer({ delayResponse: 500 })
+  const opts = lazyServer.address()
+  opts.pipelining = 2
+  const client = new Client(opts)
+  let count = 0
+
+  client.on('response', (statusCode, length, duration) => {
+    t.equal(statusCode, 200, 'status code matches')
+    t.ok(duration > 500 && duration < 600)
```

Comment:
this is probably going to make it a brittle test at some point. I understand that it's tricky to test though, but relying on timing in tests is something I would try to avoid. I am not sure how to do it though, apart from changing the way the test is written altogether

Reply:
We're doing such time based assertions in a lot of places in these tests, so I don't think this is any less appropriate than that. But if you have any clever idea on how to improve, I'd be happy to hear it.

Reply:
Not really, I think the only improvement to reduce possible brittleness is to only assert that it will take longer than 500, because setTimeout is guaranteed to trigger after and not before the specified time

Reply:
our calculations were giving ~half results, so not checking if it takes less than 600ms might not catch if the calculations in future break and start giving 2x time. We can increase it to 700 if builds start to fail.

Reply:
fair enough!